recipes/halide/build.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# On OSX, we need to ensure we're using conda's gcc/g++
+if [[ `uname` == Darwin ]]; then
+     export CC=gcc
+     export LD=gcc
+     export CXX=g++
```

Comment:
@msarahan, any thoughts on a newer LLVM? We could also try build it from a recipe here, but I'm guessing that is going to take way too long.